shbin.py

```diff
@@ -99,21 +115,42 @@ def download(url_or_path, repo, user):
     $ shbin dl https://github.com/Shiphero/pastebin/blob/main/bibo/AWS_API_fullfilment_methods/
     $ shbin dl bibo/AWS_API_fullfilment_methods/
     """
-    path = re.sub(rf"^https://github\.com/{repo.full_name}/(blob|tree)/{repo.default_branch}/", "", url_or_path)
+    full_name_repo = repo.full_name
+    pattern = r"https://github.com/(.*?)/(tree|blob)/"
+    result = re.findall(pattern, url_or_path)
+    if result:
+        full_name_repo = result[0][0]
+
+    is_from_my_repo = "https://" not in url_or_path or full_name_repo == repo.full_name
```

Comment:
a few things here. 

-  the first condition should be `not url_or_path.startswith("https://") `
- is it enough to assume that if it doesn't start with https:// then is from repo? 

- wording: "my_repo"  . I'm the user but the repo could someone else and I just have permissions, right? like mgaitan downloading from Shiphero/pastebin.   I would use `is_from_the_repo`  or something like that